contracts/minter-suite/Minters/MinterPolyptych/MinterPolyptychV0.sol

```diff
@@ -0,0 +1,704 @@
+            msg.sender ==
+                genArtCoreContract.projectIdToArtistAddress(_projectId),
+            "Only Artist"
+        );
+        _;
+    }
+
+    /**
+     * @notice Initializes contract to be a Filtered Minter for
+     * `_minterFilter`, integrated with Art Blocks core contract
+     * at address `_genArt721Address`.
+     * @param _genArt721Address Art Blocks core contract for which this
+     * contract will be a minter.
+     * @param _minterFilter Minter filter for which
+     * this will a filtered minter.
+     * @param _delegationRegistryAddress Delegation registry contract address.
+     */
+    constructor(
+        address _genArt721Address,
+        address _minterFilter,
+        address _delegationRegistryAddress
+    ) ReentrancyGuard() {
+        genArt721CoreAddress = _genArt721Address;
+        genArtCoreContract = IGenArt721CoreContractV3(_genArt721Address);
+        delegationRegistryAddress = _delegationRegistryAddress;
+        emit DelegationRegistryUpdated(_delegationRegistryAddress);
+        delegationRegistryContract = IDelegationRegistry(
+            _delegationRegistryAddress
+        );
+        minterFilterAddress = _minterFilter;
+        minterFilter = IMinterFilterV0(_minterFilter);
+        require(
+            minterFilter.genArt721CoreAddress() == _genArt721Address,
+            "Illegal contract pairing"
+        );
+    }
+
+    /**
+     *
+     * @notice Registers holders of NFTs at address `_NFTAddress` to be
+     * considered for minting. New core address is assumed to follow syntax of:
+     * `projectId = tokenId / 1_000_000`
+     * @param _NFTAddress NFT core address to be registered.
+     */
+    function registerNFTAddress(address _NFTAddress)
+        external
+        onlyCoreAdminACL(this.registerNFTAddress.selector)
+    {
+        _registeredNFTAddresses.add(_NFTAddress);
+        emit RegisteredNFTAddress(_NFTAddress);
+    }
+
+    /**
+     *
+     * @notice Unregisters holders of NFTs at address `_NFTAddress` to be
+     * considered for adding to future allowlists.
+     * @param _NFTAddress NFT core address to be unregistered.
+     */
+    function unregisterNFTAddress(address _NFTAddress)
+        external
+        onlyCoreAdminACL(this.unregisterNFTAddress.selector)
+    {
+        _registeredNFTAddresses.remove(_NFTAddress);
+        emit UnregisteredNFTAddress(_NFTAddress);
+    }
+
+    /**
+     * @notice Allows holders of NFTs with
+     * project IDs `_ownedNFTProjectIds` to mint on project `_projectId`.
+     * Allows holders of project `_ownedNFTProjectIds[0]` on token
+     * contract to mint `_projectId`.
+     * @param _projectId Project ID to enable minting on.
+     * @param _ownedNFTProjectIds Project IDs whose
+     * holders shall be allowlisted to mint project `_projectId`.
+     */
+    function allowHoldersOfProjects(
+        uint256 _projectId,
+        uint256[] memory _ownedNFTProjectIds
+    ) public onlyArtist(_projectId) {
```

Comment:
Not quite sure why we don't include an array of `address[] memory _ownedNFTAddresses` in this function (like we do on MinterHolderV2).

I **think** this minter might be trying to enforce a polyptych minter to only work with projects on the same core contract. That might be slightly over-constraining, and could result in polyptych minting for projects disconnected in time potentially impossible. For example, if a core contract has to be deprecated (i.e. no new projects due to EVM versioning in the future), polyptych minting would not be possible for the projects on the deprecated core contract.

If only allowing polyptych for projects on the same core contract is desired (e.g. if we always want polyptych projects to be performed at the same time), then we should remove the `registerNFTAddress` and `unregisterNFTAddress` functions (since those become irrelevant).

After making these design choices, we should check this minter's interface for commonality with `IFilteredMinterHolderV0` to see if we can simplify any subgraph code down the line 😄